### PR TITLE
moved Event type and made compatibility changed, calendar changes

### LINKED ISSUE #41: create generic persistence interface, and a ViewModel that handles it

### PR DESCRIPTION
**Pull request by EmmChriss**

### CODE REVIEW

File: app/src/main/java/com/lab/skuld/ui/screens/NewNoteScreen.kt [hunk 1]
```diff
@@ -106,16 +60,13 @@ fun documentToEvent(document: Document): Event {
     return Event(id = "", startDate = null, endDate = null, title = document.header, checked = true, contents = contents)
 }
 
-fun documentToEventNoID(document: Document): EventNoID {
+fun documentToEventNoID(document: Document): CreateEvent {
     val contents = document.documentContents.joinToString("\n") { data -> "${data.header} ${data.value}" }
-    return EventNoID( startDate = null, endDate = null, title = document.header, checked = true, contents = contents)
```
Comment (EEtele) [inline app/src/main/java/com/lab/skuld/ui/screens/NewNoteScreen.kt:111]: sorry you had to look at this file as a whole, respect

File: app/src/main/java/com/lab/skuld/ui/screens/SettingsScreen.kt [hunk 2]
```diff
@@ -129,7 +130,7 @@ fun ImportButton() {
         val tasksJson = file.readText()
         val tasks = gson.fromJson(tasksJson, Array<Task>::class.java)
         val eventNoIDs = tasks.map { task ->
-            EventNoID(
+            CreateEvent(
```
Comment (EEtele) [inline app/src/main/java/com/lab/skuld/ui/screens/SettingsScreen.kt:138]: it won't export dates, so can we consider that as a future-proofing feature or is that my problem from now on?

File: app/src/main/java/com/lab/skuld/ui/screens/TasksScreen.kt [hunk 3]
```diff
@@ -26,12 +25,13 @@ import com.google.firebase.auth.ktx.auth
...
 import com.google.firebase.firestore.ktx.firestore
 import com.google.firebase.ktx.Firebase
+import com.lab.skuld.model.Event
 import com.lab.skuld.ui.Screen
 import com.lab.skuld.ui.UIContextViewModel
 import com.lab.skuld.ui.rememberLiveArray
 
 
-fun TaskToEvent(task: Task): Event{
+fun TaskToEvent(task: Task): Event {
```
Comment (EEtele) [inline app/src/main/java/com/lab/skuld/ui/screens/TasksScreen.kt:34]: I disagree

File: app/src/main/java/com/lab/skuld/ui/screens/NoteScreen.kt [hunk 4]
```diff
@@ -108,26 +94,16 @@ fun ShowNoteScreen(documentt: Event) {
...
            viewModel.nav.push(Screen.NewTask(documentt))
-
-
        }) {
            Text("EDIT")
-
        }
        DeleteElementDialog(documentt.id)
-
-
```
Comment (EEtele) [inline app/src/main/java/com/lab/skuld/ui/screens/NoteScreen.kt:123]: The Whitespace Massacre

**Review by EEtele — APPROVED** [review]
42

File: app/src/main/java/com/lab/skuld/ui/screens/TasksScreen.kt (hunk 3 above)
Comment (EmmChriss) [reply app/src/main/java/com/lab/skuld/ui/screens/TasksScreen.kt:34]: :))

**Pull request merged**